File: awa/awa/sdk/utils/activity/list_directory_activity.py
from datetime import timedelta
from pathlib import Path

from temporalio import workflow

from awa.sdk import constants


async def list_directory_activity(dir_path: str | Path, ignore_file_path: str | Path | None = None) -> list[str]:
    """List the contents of a directory.

    Args:
        dir_path: The path to the directory to list. Can be a string or Path object.
        ignore_file_path: Optional path to an ignore file (like .gitignore) to exclude files.
            Can be a string or Path object. Defaults to None.

    Returns:
        list[str]: A list of file and directory names in the specified directory.

    """
    return await workflow.execute_activity(
        constants.ACTIVITY_LIST_DIRECTORY,
        args=[str(dir_path), str(ignore_file_path) if ignore_file_path else None],
        task_queue=constants.AWA_DEFAULT_TASK_QUEUE,
        start_to_close_timeout=timedelta(seconds=constants.FILE_IO_TIMEOUT_SECONDS),
    )

File: awa/sdk_dist/csharp/Awa.Client/Utilities/Activity/ListDirectoryActivity.cs
using System;
using System.Threading.Tasks;
using Awa.Client.Constants;

namespace Awa.Client.Utilities.Activity
{
    public static partial class ActivityUtilities
    {
        /// <summary>
        /// List the contents of a directory.
        /// </summary>
        /// <param name="dirPath">The path to the directory to list.</param>
        /// <param name="ignoreFilePath">Optional path to an ignore file (like .gitignore) to exclude files. Defaults to null.</param>
        /// <returns>A list of file and directory names in the specified directory.</returns>
        public static async Task<string[]?> ListDirectoryActivity(string dirPath, string? ignoreFilePath = null)
        {
            return await Temporalio.Workflows.Workflow.ExecuteActivityAsync<string[]?>(
                AwaConstants.ActivityListDirectory,
                new object?[] { dirPath, ignoreFilePath },
                new()
                {
                    TaskQueue = AwaConstants.AwaDefaultTaskQueue,
                    StartToCloseTimeout = TimeSpan.FromSeconds(AwaConstants.FileIoTimeoutSeconds)
                });
        }
    }
}
